src/test/ui/async-await/async-block-control-flow-static-semantics.stderr

```diff
@@ -29,13 +29,13 @@ LL | |
 LL | | }
    | |_^ expected `u8`, found `()`
 
-error[E0271]: expected `impl Future<Output = u8>` to be a future that resolves to `()`, but it resolves to `u8`
+error[E0271]: expected `[static `async` block@$DIR/async-block-control-flow-static-semantics.rs:23:17: 25:6]` to be a future that resolves to `()`, but it resolves to `u8`
```

Comment:
I think the nested backticks in ``` `async` ``` is a bit strange looking in these type errors.

Also, I'm not sure if we expect users to know what `static` generator movability is -- if the generator kind is `Async`, maybe we could just drop the `static` (maybe unless we pass in `-Zverbose` or something)?

Reply:
Agreed, I pushed an updated version that does this.

Also now uses the `descr()` method for explicit diagnostics that still print `` `async` block``, but when printing the type of the block itself, the output is cleaner now.